k9mail-library/src/main/java/com/fsck/k9/mail/internet/ListHeaders.java

```diff
@@ -0,0 +1,46 @@
+package com.fsck.k9.mail.internet;
+
+import android.net.MailTo;
+
+import com.fsck.k9.mail.Address;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Intended to cover:
+ *
+ * RFC 2369
+ * The Use of URLs as Meta-Syntax for Core Mail List Commands
+ * and their Transport through Message Header Fields
+ * https://www.ietf.org/rfc/rfc2369.txt
+ *
+ * This is the following fields:
+ *
+ * List-Help
+ * List-Subscribe
+ * List-Unsubscribe
+ * List-Post
+ * List-Owner
+ * List-Archive
+ *
+ * Currently only provides a utility method for List-Post
+ **/
+public class ListHeaders {
+    public static final String LIST_POST_HEADER = "List-Post";
+    private static final Pattern mailtoContainerPattern = Pattern.compile("<(.+)>");
+
+    public static Address[] parsePostAddress(String headerValue) {
+        if (headerValue == null || headerValue.isEmpty()) {
+            return new Address[]{};
+        } else {
```

Comment:
To avoid increasing the indentation depth you can remove the redundant `else`. The action in Android Studio is called `Remove redundant 'else'`.

Example:
```java
if (headerValue == null || headerValue.isEmpty()) {
    return new Address[0];
}

Matcher matcher = mailtoContainerPattern.matcher(headerValue);
if (!matcher.find()) {
    return new Address[0];
}

String mailtoUri = matcher.group(1);
String emailAddress = MailTo.parse(mailtoUri).getTo();
return new Address[] { new Address(emailAddress) };
```

(Deeply) nested expressions make the code harder to read. That's why I introduced some local variables.